azure-pipelines.yml

```diff
@@ -171,6 +171,7 @@ jobs:
     dependsOn: [linting, git_commit, Ubuntu_Jammy_Jellyfish]
     condition: |
       and(
+        succeeded(),
         in(dependencies['Ubuntu_Jammy_Jellyfish']['result'], 'Succeeded', 'Skipped'),
```

Comment:
This change was due to https://github.com/scikit-learn/scikit-learn/pull/21497. This job wants to run if the dependencies succeeded or skipped.

From https://github.com/MicrosoftDocs/azure-devops-docs/issues/7738#issuecomment-611815486, I think the workaround is:

```suggestion
        # Runs when dependencies succeeded or skipped
        not(or(failed(), canceled())
```

I am not sure if that is enough to get Azure to cancel running jobs.

If this is too complex, I'm okay with reverting #21497.

Reply:
Indeed, I did that too quickly. Let's try yours.